### DISCUSSION TITLE
regarding firmware versions

### DISCUSSION THREAD

**Discussion by ExileFox** [dmitry-s93/MControlCenter#193] [Q&A] [answered]
I noticed that there is an entry for a GF65 Thin 10UE on the list but the firmware is different.
Like I have the same model, but if i pull the same string, i get **E16W2IMS.105** which is slightly different than **16W2EMS1.101**
Fan control, cooler boost both work just as described in this case.

But let's theorize about a model where two different firmwares dictate if a function is working as intended/available i.e. Keyboard backlit.

How should different firmwares should be handled?

**Comment by mutchiko** [comment, marked as answer]
you are comparing BIOS version to EC version, which is not the same.

a good read on [EC naming](https://github.com/BeardOverflow/msi-ec/blob/main/docs/device_support_guide.md#additional-info) if you're still interested

**Comment by ExileFox** [comment]
Makes sense but pretty easy to miss. Either way I have switched laptop to a Gigabyte so I no longer have any use for this software.
